### PR TITLE
🕸️  Fix Active Program Path

### PR DESCRIPTION
**Pull request by demariadaniel**
Separate URL Navigation and Menu Item Open / Close
Use URL Program short name to determine initial state

Outcome of discussion on https://github.com/icgc-argo/rdpc-ui/pull/114

### CODE REVIEW

File: src/app/(post-login)/submission/components/ProgramMenu.tsx [hunk 1]
```diff
@@ -67,12 +67,11 @@ const StatusMenuItem: FC<{ children: ReactNode }> = ({ children }) => {
 const ProgramMenu = ({ shortNameSearchQuery }: { shortNameSearchQuery: string }) => {
 	const params = useParams();
 	const pathname = usePathname();
-	const router = useRouter();
 	const { egoJwt } = useAuthContext();
 	const { DATA_CENTER } = useAppConfigContext();
 
 	// params can be arrays from dynamic routing
-	const activeProgramName = typeof params.shortName === 'string' ? params.shortName : '';
+	const activeProgramRoute = typeof params.shortName === 'string' ? params.shortName : '';
```
Comment (demariadaniel) [inline src/app/(post-login)/submission/components/ProgramMenu.tsx:74]: URL params still determine initial state of page, functionality can be added

File: src/app/(post-login)/submission/components/ProgramMenu.tsx [hunk 2]
```diff
@@ -86,16 +85,9 @@ const ProgramMenu = ({ shortNameSearchQuery }: { shortNameSearchQuery: string })
 
 	const sortedProgramList = orderBy(programs, 'shortName');
 
-	const setActiveProgram =
```
Comment (demariadaniel) [inline src/app/(post-login)/submission/components/ProgramMenu.tsx:89]: No longer change URL when clicking menu item

File: src/app/(post-login)/submission/components/ProgramMenu.tsx [hunk 3]
```diff
@@ -86,16 +85,9 @@ const ProgramMenu = ({ shortNameSearchQuery }: { shortNameSearchQuery: string })
...
-		(shortName?: string): MouseEventHandler =>
-		() => {
-			if (shortName) {
-				const url = `/submission/program/${shortName}/registration`;
-				router.push(url);
-			}
-		};
+	const [activeProgram, setActiveProgram] = useState<string | undefined>(activeProgramRoute);
```
Comment (demariadaniel) [inline src/app/(post-login)/submission/components/ProgramMenu.tsx:88]: Local state manages which Menu Item is open and initial value comes from URL

File: src/app/(post-login)/submission/components/ProgramMenu.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (demariadaniel) [reply src/app/(post-login)/submission/components/ProgramMenu.tsx:74]: Yeah that's a good idea, was trying to disambiguate between the value in the URL and the currently open menu item

File: src/app/(post-login)/submission/components/ProgramMenu.tsx [hunk 4]
```diff
@@ -251,9 +250,11 @@ const renderSubmissionStatusIcon = (
...
 	programName,
 	userRoles,
+	isActive,
 }: {
 	programName: string;
 	userRoles: UserRoleList;
+	isActive: boolean;
 }) => {
 	const pathname = usePathname();
 	const pathnameLastSegment = pathname.split('/').at(-1);
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply src/app/(post-login)/submission/components/ProgramMenu.tsx:259]: Oops, removed

File: src/app/(post-login)/submission/components/ProgramMenu.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ciaranschutte) [reply src/app/(post-login)/submission/components/ProgramMenu.tsx]: Is great!

[the remark this answers is not in the document]
Comment (demariadaniel) [reply src/app/(post-login)/submission/components/ProgramMenu.tsx]: Updated

**Review by ciaranschutte — APPROVED** [review]

**Pull request merged**